### PR TITLE
Feature/d3 asim 2348: Adding infinitebus/market_maker to backend plots.

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

File: src/d3a/setup/d3a-settings.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
       "ENERGY_RATE_DECREASE_PER_UPDATE": 1,
       "SETUP_FILE_PATH": null,
       "EXPORT_DEVICE_PLOTS": true,
-      "EXPORT_ENERGY_TRADE_PROFILE_HR": false,
+      "HIGH_RESOLUTION_PLOTS": false,
```
Comment (hannesdiedrich) [inline src/d3a/setup/d3a-settings.json:21]: Pleas see my comment in d3a-interface for this.

File: setup.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
         # TODO: Workaround for https://github.com/ethereum/py-solc/issues/64
         REQUIREMENTS.extend(
             ['d3a-interface @ '
-             f'git+https://github.com/gridsingularity/d3a-interface.git',
+             f'git+https://github.com/gridsingularity/d3a-interface.git@feature/D3ASIM-2348',
```
Comment (hannesdiedrich) [inline setup.py:15]: please revert before merging.

File: src/d3a/d3a_core/sim_results/device_statistics.py [hunk 3]
```diff
@@ -152,7 +153,14 @@ def _gather_device_statistics(cls, area: Area, subdict: Dict, flat_result_dict:
 
         elif isinstance(area.strategy, FinitePowerPlant):
             market = list(area.parent.past_markets)[-1]
-            _create_or_append_dict(subdict, "production_kWh",
+            _create_or_append_dict(subdict, "energy_buffer_kWh",
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/sim_results/device_statistics.py:156]: Can you please explain your decision for this name? Thanks!

File: src/d3a/d3a_core/sim_results/device_statistics.py [hunk 4]
```diff
@@ -59,7 +60,7 @@ def _device_price_stats(cls, area: Area, subdict: Dict):
         trade_price_list = []
         for t in market.trades:
             if t.seller == area.name or t.buyer == area.name:
-                trade_price_list.append(t.offer.price / 100.0 / t.offer.energy)
+                trade_price_list.append(t.offer.energy_rate)
```
Comment (spyrostz) [inline src/d3a/d3a_core/sim_results/device_statistics.py:63]: Energy rate that the `trade_price_list` expects is in euros/kWh, while energy rate of the offer is in cents/kWh. I think the /100.0 is still required here.

File: src/d3a/d3a_core/sim_results/device_statistics.py (hunk 3 above)
Comment (faizan2590) [reply src/d3a/d3a_core/sim_results/device_statistics.py]: You are right, distinguishing between finite and infinite power sources and using `energy_buffer_kWh ` only for `InfiniteBus` and `CommercialProducer` strategy.

**Pull request merged**